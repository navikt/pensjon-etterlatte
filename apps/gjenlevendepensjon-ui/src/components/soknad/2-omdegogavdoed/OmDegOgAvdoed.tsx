import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoekerOgAvdoed } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { FormProvider, useForm } from "react-hook-form";
import { RHFInput } from "../../felles/rhf/RHFInput";
import { SkjemaGruppe } from "../../felles/SkjemaGruppe";
import ForholdTilAvdoedeSkjema from "./forholdTilAvdoede/ForholdTilAvdoedeSkjema";
import Feilmeldinger from "../../felles/Feilmeldinger";
import Datovelger from "../../felles/Datovelger";
import { Cell, Grid, Label, Heading } from "@navikt/ds-react";
import NySivilstatus from "./nySivilstatus/NySivilstatus";
import Navigasjon from "../../felles/Navigasjon";
import { deepCopy } from "../../../utils/deepCopy";
import {SkjemaElement} from "../../felles/SkjemaElement";

const OmDegOgAvdoed: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<ISoekerOgAvdoed>({
        defaultValues: state.omDegOgAvdoed || {},
        shouldUnregister: true,
    });

    const {
        handleSubmit,
        formState: { errors },
        getValues,
    } = methods;

    const erValidert = state.omDegOgAvdoed.erValidert;

    const lagreNeste = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(data), erValidert: true } });
        neste!!();
    };

    const lagreTilbake = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(data), erValidert: true } });
        forrige!!();
    };

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues();
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(verdier), erValidert: false } });
        forrige!!();
    };

    return (
        <>
            <SkjemaElement>
                <Heading size={"medium"} className={"center"}>
                    {t("omDegOgAvdoed.tittel")}
                </Heading>
            </SkjemaElement>

            <FormProvider {...methods}>
                <form>
                    <SkjemaGruppe>
                        <Label>{t("omDegOgAvdoed.avdoed.hvem")}</Label>

                        <Grid>
                            <Cell xs={12} md={6}>
                                <RHFInput name={"avdoed.fornavn"} label={t("omDegOgAvdoed.avdoed.fornavn")} />
                            </Cell>

                            <Cell xs={12} md={6}>
                                <RHFInput name={"avdoed.etternavn"} label={t("omDegOgAvdoed.avdoed.etternavn")} />
                            </Cell>
                        </Grid>
                    </SkjemaGruppe>

                    <SkjemaGruppe>
                        <Label>{t("omDegOgAvdoed.avdoed.datoForDoedsfallet")}</Label>

                        <Datovelger
                            name={"avdoed.datoForDoedsfallet"}
                            label={t("omDegOgAvdoed.avdoed.dato")}
                            maxDate={new Date()}
                        />
                    </SkjemaGruppe>

                    <ForholdTilAvdoedeSkjema />

                    <NySivilstatus />

                    <Feilmeldinger errors={errors} />

                    <Navigasjon
                        forrige={{
                            onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering,
                        }}
                        neste={{ onClick: handleSubmit(lagreNeste) }}
                    />
                </form>
            </FormProvider>
        </>
    );
};

export default OmDegOgAvdoed;
