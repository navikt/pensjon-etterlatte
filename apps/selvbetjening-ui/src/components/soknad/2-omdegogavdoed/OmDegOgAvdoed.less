@import "../../../App.less";

.opplysninger {
    .typo-element,
    .typo-normal {
        font-size: 1.1rem;
    }

    .typo-normal {
        margin: 0.2rem 0 1rem 0;
    }
}
