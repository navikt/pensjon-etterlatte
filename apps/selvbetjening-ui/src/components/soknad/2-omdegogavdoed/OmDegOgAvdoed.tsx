import "../../../App.less";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoekerOgAvdoed } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { FormProvider, useForm } from "react-hook-form";
import { Element, Systemtittel } from "nav-frontend-typografi";
import { RHFInput } from "../../felles/RHFInput";
import { SkjemaGruppe } from "nav-frontend-skjema";
import ForholdTilAvdoedeSkjema from "./forholdTilAvdoede/ForholdTilAvdoedeSkjema";
import Feilmeldinger from "../../felles/Feilmeldinger";
import Datovelger from "../../felles/Datovelger";
import { Hovedknapp, Knapp } from "nav-frontend-knapper";
import AlertStripe from "nav-frontend-alertstriper";
import { RHFRadio } from "../../felles/RHFRadio";
import { IValg } from "../../../typer/Spoersmaal";
import NySivilstatus from "./forholdTilAvdoede/nySivilstatus/NySivilstatus";

const OmDegOgAvdoed: SoknadSteg = ({ neste, forrige }) => {
    const { t, i18n } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const lagre = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: data })
        neste!!()
    };

    const methods = useForm<ISoekerOgAvdoed>({
        defaultValues: state.omDegOgAvdoed || {},
        shouldUnregister: true
    });

    const {
        handleSubmit,
        watch,
        formState: { errors }
    } = methods;

    const dtf = Intl.DateTimeFormat(i18n.language, { day: "numeric", month: "long", year: "numeric" });

    const datoForDoedsfallet = watch("avdoed.datoForDoedsfallet") || undefined;

    let foersteFraDato;
    if (!!datoForDoedsfallet) {
        const dato = new Date(datoForDoedsfallet)
        foersteFraDato = dtf.format(new Date(dato.getFullYear(), dato.getMonth() + 1, 1))
    }

    return (
        <>
            {/* Steg 2 */}
            <SkjemaGruppe>
                <Systemtittel className={"center"}>
                    Om deg og avdøde
                </Systemtittel>
            </SkjemaGruppe>

            {/* Skjema for utfylling av info om innlogget bruker / søker */}
            <FormProvider {...methods}>
                <form>
                    <Element>Hvem er det som er død?</Element>
                    <div className={"rad skjemagruppe"}>
                        <div className={"kol-50"}>
                            <RHFInput
                                name={"avdoed.fornavn"}
                                label={"Fornavn"}
                                // TODO: Validere telefon ... ?
                            />
                        </div>

                        <div className={"kol-50"}>
                            {/* 2.5 */}
                            <RHFInput
                                name={"avdoed.etternavn"}
                                label={"Etternavn"}
                                // TODO: Validere e-post
                            />
                        </div>
                    </div>

                    <SkjemaGruppe>
                        <Element>Når skjedde dødsfallet</Element>

                        <Datovelger
                            name={"avdoed.datoForDoedsfallet"}
                            label={"Dato"}
                        />

                        {foersteFraDato && (
                            <AlertStripe
                                type={"info"}
                                form={"inline"}
                            >
                                Du kan ha rett på gjenlevendepensjon fra {foersteFraDato}
                            </AlertStripe>
                        )}
                    </SkjemaGruppe>

                    <SkjemaGruppe>
                        <RHFRadio
                            name={"avdoed.doedsfallAarsak"}
                            legend={"Skyldes dødsfallet yrkesskade/yrkessykdom?"}
                            radios={[
                                { label: IValg.JA, value: IValg.JA },
                                { label: IValg.NEI, value: IValg.NEI },
                                { label: IValg.VET_IKKE, value: IValg.VET_IKKE }
                            ]}
                        />
                    </SkjemaGruppe>

                    {/* 2.9 */}
                    <ForholdTilAvdoedeSkjema />

                    <NySivilstatus />

                    <br />

                    <Feilmeldinger errors={errors} />

                    <SkjemaGruppe className={"navigasjon-rad"}>
                        <Knapp htmlType={"button"} onClick={forrige}>
                            {t("knapp.tilbake")}
                        </Knapp>

                        <Hovedknapp htmlType={"button"} onClick={handleSubmit(lagre)}>
                            {t("knapp.neste")}
                        </Hovedknapp>
                    </SkjemaGruppe>
                </form>
            </FormProvider>
        </>
    );
};

export default OmDegOgAvdoed;
