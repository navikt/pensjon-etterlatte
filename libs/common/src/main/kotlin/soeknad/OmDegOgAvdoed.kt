package soeknad

import com.fasterxml.jackson.annotation.JsonValue
import no.nav.etterlatte.libs.common.soeknad.Valg


data class OmDegOgAvdoed(
    val avdoed: Avdoed,
    val forholdTilAvdoede: ForholdTilAvdoede,
    val nySivilstatus: NySivilstatus
)

data class Avdoed(
    val datoForDoedsfallet: String,
    val etternavn: String,
    val fornavn: String
)

data class ForholdTilAvdoede(
    val relasjon: ForholdTilAvdoedeType,
    val datoForInngaattPartnerskap: String?,
    val datoForSkilsmisse: String?,
    val datoForInngaattSamboerskap: String?,
    val datoForSamlivsbrudd: String?,
    val fellesBarn: Valg?,
    val samboereMedFellesBarn: Valg?,
    val omsorgForBarn: Valg?,
    val tidligereGift: Valg?,
    val mottokBidrag: Valg?,
    val mottokEktefelleBidrag: Valg?
)

enum class ForholdTilAvdoedeType(@get:JsonValue val relasjon: String) {
    gift("avdoede.relasjon.gift"),
    separert("avdoede.relasjon.separert"),
    samboer("avdoede.relasjon.samboer"),
    skilt("avdoede.relasjon.skilt"),
    tidligereSamboer("avdoede.relasjon.tidligereSamboer")
}
