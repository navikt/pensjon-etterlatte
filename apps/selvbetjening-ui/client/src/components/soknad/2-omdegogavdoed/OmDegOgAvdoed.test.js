import { act, render, fireEvent } from "@testing-library/react";
import * as JSutils from "nav-frontend-js-utils";
import OmDegOgAvdoed from "./OmDegOgAvdoed";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => ({
        t: (str) => str,
        i18n: {
            changeLanguage: () => new Promise(() => {}),
        },
    }),
}));

JSutils.guid = jest.fn(() => "123");

jest.mock("uuid", () => ({
    v4: () => "456",
}));

describe("Om deg og avdÃ¸d", () => {
    it("Snapshot uten samboerskjema", () => {
        const { container } = render(<OmDegOgAvdoed />);
        expect(container).toMatchSnapshot();
    });

    it("Snapshot med sivilstatus samboer", () => {
        const { container, getByText, getByLabelText } = render(<OmDegOgAvdoed />);

        act(() => {
            fireEvent.click(getByLabelText("nySivilstatus.samboerskap"));
        });
        expect(getByText("omDegOgAvdoed.nySivilstatus.samboerskap.hattBarnEllerVaertGift")).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
