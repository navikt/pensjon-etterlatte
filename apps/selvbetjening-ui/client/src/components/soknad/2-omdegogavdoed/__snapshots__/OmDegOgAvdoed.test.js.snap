// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om deg og avd√∏d Snapshot med sivilstatus samboer 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </p>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <label
                class="skjemaelement__label"
                for="avdoed.fornavn"
              >
                omDegOgAvdoed.avdoed.fornavn
              </label>
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <label
                class="skjemaelement__label"
                for="avdoed.etternavn"
              >
                omDegOgAvdoed.avdoed.etternavn
              </label>
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </p>
      <section
        class="skjemaelement undefined"
      >
        <label
          class="skjemaelement__label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato felles.datoformat
        </label>
        <div
          class="datovelger"
        >
          <div>
            <div
              class="ds-datepicker"
            >
              <div
                class="ds-datepicker"
              >
                <div
                  class="ds-datepicker__input navds-form-field navds-form-field--medium"
                >
                  <label
                    class="navds-text-field__label sr-only navds-label"
                    for="avdoed.datoForDoedsfallet"
                  >
                    abc
                  </label>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    autocorrect="off"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="avdoed.datoForDoedsfallet"
                    inputmode="text"
                    maxlength="10"
                    name="avdoed.datoForDoedsfallet"
                    placeholder="felles.datoEksempel"
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    id="textField-error-456"
                  />
                </div>
                <button
                  aria-expanded="false"
                  class="ds-datepicker__kalenderknapp"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Kalender
                  </span>
                  <span
                    aria-hidden="true"
                    class="ds-datepicker__kalenderknapp__icon"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      role="presentation"
                      style="pointer-events: none;"
                      viewBox="0 0 18 18"
                      width="16"
                    >
                      <title>
                        Kalenderikon
                      </title>
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        stroke="#0067C5"
                      >
                        <path
                          d="M4 2.667H1.333v14h15.334v-14H14"
                        />
                        <path
                          d="M4 1.333h2V4H4zm8 0h2V4h-2zM6 2h6M1.333 6h15.334"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        
      </section>
      <div
        aria-live="polite"
      />
    </fieldset>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.forholdTilAvdoede.relasjon
        </legend>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.gift"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.gift
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.separert"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.separert
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.samboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.samboer
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.skilt"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.skilt
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.tidligereSamboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.tidligereSamboer
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe"
        >
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ingen"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ingen
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ekteskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ekteskap
              </span>
            </label>
            <label
              class="inputPanel radioPanel inputPanel--checked"
            >
              <input
                aria-checked="true"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.samboerskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.samboerskap
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        class="navds-panel navds-panel--border"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe"
        >
          <h1
            class="navds-heading navds-heading--small"
          >
            omDegOgAvdoed.nySivilstatus.samboerskap.samboer.tittel
          </h1>
          <div
            aria-live="polite"
          />
        </fieldset>
        <div
          class="navds-grid"
        >
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
          >
            <div
              class="kol-50"
            >
              <div
                class="skjemaelement"
              >
                <label
                  class="skjemaelement__label"
                  for="nySivilstatus.samboerskap.samboer.fornavn"
                >
                  omDegOgAvdoed.nySivilstatus.samboerskap.samboer.fornavn
                </label>
                <input
                  aria-invalid="false"
                  class="skjemaelement__input input--fullbredde"
                  id="nySivilstatus.samboerskap.samboer.fornavn"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
          >
            <div
              class="kol-50"
            >
              <div
                class="skjemaelement"
              >
                <label
                  class="skjemaelement__label"
                  for="nySivilstatus.samboerskap.samboer.etternavn"
                >
                  omDegOgAvdoed.nySivilstatus.samboerskap.samboer.etternavn
                </label>
                <input
                  aria-invalid="false"
                  class="skjemaelement__input input--fullbredde"
                  id="nySivilstatus.samboerskap.samboer.etternavn"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="skjemaelement"
        >
          <label
            class="skjemaelement__label"
            for="nySivilstatus.samboerskap.samboer.foedselsnummer"
          >
            felles.fnr
          </label>
          <input
            aria-invalid="false"
            class="skjemaelement__input input--s"
            id="nySivilstatus.samboerskap.samboer.foedselsnummer"
            placeholder="felles.fnrPlaceholder"
            required=""
            type="tel"
            value=""
          />
        </div>
        <div
          id="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
        >
          <fieldset
            aria-invalid="false"
            class="skjemagruppe inputPanelGruppe inline"
          >
            <legend
              class="skjemagruppe__legend"
            >
              omDegOgAvdoed.nySivilstatus.samboerskap.hattBarnEllerVaertGift
            </legend>
            <div
              class="inputPanelGruppe__inner"
            >
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                  required=""
                  type="radio"
                  value="radiobuttons.ja"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.ja
                </span>
              </label>
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                  required=""
                  type="radio"
                  value="radiobuttons.nei"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.nei
                </span>
              </label>
            </div>
            <div
              aria-live="polite"
            />
          </fieldset>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.neste
          </span>
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.avbryt
        </span>
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;

exports[`Om deg og avd√∏d Snapshot uten samboerskjema 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </p>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <label
                class="skjemaelement__label"
                for="avdoed.fornavn"
              >
                omDegOgAvdoed.avdoed.fornavn
              </label>
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <label
                class="skjemaelement__label"
                for="avdoed.etternavn"
              >
                omDegOgAvdoed.avdoed.etternavn
              </label>
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </p>
      <section
        class="skjemaelement undefined"
      >
        <label
          class="skjemaelement__label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato felles.datoformat
        </label>
        <div
          class="datovelger"
        >
          <div>
            <div
              class="ds-datepicker"
            >
              <div
                class="ds-datepicker"
              >
                <div
                  class="ds-datepicker__input navds-form-field navds-form-field--medium"
                >
                  <label
                    class="navds-text-field__label sr-only navds-label"
                    for="avdoed.datoForDoedsfallet"
                  >
                    abc
                  </label>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    autocorrect="off"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="avdoed.datoForDoedsfallet"
                    inputmode="text"
                    maxlength="10"
                    name="avdoed.datoForDoedsfallet"
                    placeholder="felles.datoEksempel"
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    id="textField-error-456"
                  />
                </div>
                <button
                  aria-expanded="false"
                  class="ds-datepicker__kalenderknapp"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Kalender
                  </span>
                  <span
                    aria-hidden="true"
                    class="ds-datepicker__kalenderknapp__icon"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      role="presentation"
                      style="pointer-events: none;"
                      viewBox="0 0 18 18"
                      width="16"
                    >
                      <title>
                        Kalenderikon
                      </title>
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        stroke="#0067C5"
                      >
                        <path
                          d="M4 2.667H1.333v14h15.334v-14H14"
                        />
                        <path
                          d="M4 1.333h2V4H4zm8 0h2V4h-2zM6 2h6M1.333 6h15.334"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        
      </section>
      <div
        aria-live="polite"
      />
    </fieldset>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.forholdTilAvdoede.relasjon
        </legend>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.gift"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.gift
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.separert"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.separert
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.samboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.samboer
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.skilt"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.skilt
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.tidligereSamboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.tidligereSamboer
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe"
        >
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ingen"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ingen
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ekteskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ekteskap
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.samboerskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.samboerskap
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.neste
          </span>
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.avbryt
        </span>
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;
